### ISSUE TITLE
`onSelectionChange` event

### ISSUE THREAD

**Issue by devongovett** [facebook/react#5785]
The `onSelect` event does not fire while the user is dragging the mouse, only after they release. It would be nice to have an event that fired immediately whenever the selection changed, including while dragging the mouse. This is useful, for example, when syncing selections between multiple collaborators, ala google docs.

The native [selectionchange](https://developer.mozilla.org/en-US/docs/Web/Events/selectionchange) event is supported cross browser, way back in IE, Chrome, and Safari, and recently in Firefox 43. Is this something React could polyfill and expose to components?

**Comment by YurkaninRyan** [comment]
Do any of the more seasoned contributors have any opinions on what this should look like?  Should it just become a feature of `onSelect` or does it merit it's own event hook?

It seems like the current React `onSelect` is a bit of a mixture of the two already.

**Comment by gaearon** [comment]
Seems reasonable to me to mirror the browser API.

**Comment by flarnie** [comment]
I'd like to look into this because it could affect Draft.js. Will post more when I get time.

**Comment by raphasilvac** [comment]
Hello @flarnie,
I think it would be nice for Draft.js

I think there is a problem on Draft.js, 
the selectionState is only updated after "mouseup", 
if you keep mouse pressed and paste or write some word you will see that the word or some part of it goes to a wrong place ("previous selectionState").

This is reproducible on Draft.js home/demo page and on facebook status field.

I think it is not a good solution, but as a workaround I added the following code on Draft.js componentDidMount:
`document.addEventListener("selectionchange", this._onSelect);`

In my tests it works, but maybe you can know some unexpected effect about it.

Sorry about my english!

Thank you.

**Comment by ianstormtaylor** [comment]
@flarnie any word on this? We'd love this for Slate.js as well!

**Comment by kedromelon** [comment]
came across this PR and noticed it hadn't been linked here yet -- for documentation's sake, it looks like a attempt at this was made in #10746. @gaearon mentioned that it may be best to expose the event, but not add a polyfill.

**Comment by ianstormtaylor** [comment]
Not stale!

This is still a pain point that's lacking in React's API. I'm not sure what the solution is since it seems like no new events are being added to React despite them being standardized?

**Comment by TrySound** [comment]
This will probably be a solution
https://github.com/facebook/react/pull/17651

**Comment by ianstormtaylor** [comment]
Stil not stale...

**Comment by Sudobeast** [comment]
bump

**Comment by Fortidude** [comment]
any ideas someone? : )

**Comment by b5414** [comment]
🙂
